src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/GOAGeneratorUtilities.java

```diff
@@ -1,102 +1,139 @@
+     * S. aureus, and T. gondii).
+     *
+     * Returns a String describing the first issue found or an empty String if there is no issue.
+     *
+     * @param protein Protein to check for issue disqualifying it for using in GAF annotation
+     * @return String describing first issue found or an empty String if no issue.
+     * @throws Exception Thrown if there is a problem in retrieving information from the database for species or
+     * reference database
      */
-    public static boolean isValidProtein(GKInstance referenceEntityInst, GKInstance speciesInst) throws Exception {
-        if (referenceEntityInst != null && speciesInst != null) {
-            GKInstance referenceDatabaseInst = (GKInstance) referenceEntityInst.getAttributeValue(ReactomeJavaConstants.referenceDatabase);
-            if (referenceDatabaseInst != null && referenceDatabaseInst.getDisplayName().equals(UNIPROT_STRING) && speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference) != null) {
-                return true;
-            }
+    public static String getAnyIssueForAnnotationDisqualification(GKInstance protein) throws Exception {
+        String proteinIssue = "";
+        if (!isValidProtein(protein)) {
+            proteinIssue = protein.getExtendedDisplayName() + " is an invalid protein, skipping GO annotation";
+        } else if (hasExcludedMicrobialSpecies(protein)) {
+            proteinIssue = protein.getExtendedDisplayName() + " is from an excluded microbial species, " +
+                "skipping GO annotation";
         }
-        return false;
-    }
 
-    // This method checks the validity of the PhysicalEntity of a Catalyst instance by checking it has a compartment attribute.
+        return proteinIssue;
+    }
 
     /**
-     * Shared catalyst validation method between MolecularFunction and BiologicalProcess classes that checks for existence of compartment attribute.
-     * @param catalystPEInst -- PhysicalEntity attribute from a Catalyst instance
-     * @return -- true/false indicating PhysicalEntity validity.
-     * @throws Exception -- MySQLAdaptor exception.
+     * Retrieves the Reference Entity for the Protein.
+     * @param protein Protein for which to retrieve the reference entity
+     * @return GKInstance representing the protein's reference entity (or null if the protein has no reference entity)
+     * @throws Exception Thrown if there is a problem in retrieving the protein's reference entity from the database
      */
-    public static boolean isValidCatalystPE(GKInstance catalystPEInst) throws Exception {
-        return catalystPEInst != null && catalystPEInst.getAttributeValue(ReactomeJavaConstants.compartment) != null;
+    public static GKInstance getReferenceEntityFromProtein(GKInstance protein) throws Exception {
+         return (GKInstance) protein.getAttributeValue(ReactomeJavaConstants.referenceEntity);
```

Comment:
This method just calls getAttributeValue, so I'm not sure this needs to be its own method. But... the name suggests that it only works for proteins, so maybe this method should have a check that `protein` is _actually_ a protein?

Reply:
Added a check to throw an `IllegalArgumentException` if the argument is not a protein.